pkg/metrics/pushgateway.go

```diff
@@ -0,0 +1,88 @@
+package metrics
+
+import (
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	namespace     = os.Getenv("NAMESPACE")
+	podName       = os.Getenv("POD_NAME")
+	promNamespace = "vault_creds"
+
+	errorTime = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "last_renewal_error_timestamp_seconds",
+		Help:      "The timestamp of the last error during renewal of a secret",
+	})
+
+	errorCount = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: promNamespace,
+		Name:      "error_count",
```

Comment:
`credential_renewal_errors_total` would be more in line with prom naming conventions